torch/_tensor.py

```diff
@@ -106,7 +106,7 @@ def __deepcopy__(self, memo):
                     not torch._C._has_storage(self)
                     and self.device.type == torch._C._get_privateuse1_backend_name()
                 )
-                or (type(self) is not Tensor and self.data_ptr() == 0)
+                or (type(self) is not Tensor)
```

Comment:
This is very much not ok. Will answer on the issue.

Reply:
i change this to satisfy the deepcopy of the DTensor, i found that the origin DTensor is go through by  is not Tensor and data_ptr  == 0 , but now we set the data_ptr to its wrapper local_tensor ptr, it think this should be change. i can add a new condition ```isinstance(self, DTensor)``` to check it?

Reply:
Special casing DTensor here is not really ok either no.